app/src/main/kotlin/com/bitwarden/authenticator/ui/authenticator/feature/itemlisting/ItemListingViewModel.kt

```diff
@@ -403,101 +411,87 @@ class ItemListingViewModel @Inject constructor(
-                clearDialogState = true,
-            )
-        } else {
+        if (localItems.isEmpty()) {
+            // If there are no local items, show empty state:
             mutableStateFlow.update {
                 it.copy(
-                    viewState = ItemListingState.ViewState.Error(
-                        R.string.generic_error_message.asText(),
+                    viewState = ItemListingState.ViewState.NoItems(
+                        actionCard = action.sharedCodesState.toActionCard(),
                     ),
-                    dialog = null,
                 )
             }
+        } else {
+            val viewState = ItemListingState.ViewState.Content(
+                favoriteItems = localItems
+                    .filter { it.source is AuthenticatorItem.Source.Local && it.source.isFavorite }
+                    .map {
+                        it.toDisplayItem(alertThresholdSeconds = state.alertThresholdSeconds)
+                    },
+                itemList = localItems
+                    .filter { it.source is AuthenticatorItem.Source.Local && !it.source.isFavorite }
+                    .map {
+                        it.toDisplayItem(alertThresholdSeconds = state.alertThresholdSeconds)
+                    },
+                actionCard = action.sharedCodesState.toActionCard(),
+            )
+            mutableStateFlow.update { it.copy(viewState = viewState) }
         }
     }
 
-    private fun authenticatorDataLoadedReceive(
-        authenticatorData: DataState.Loaded<List<VerificationCodeItem>>,
-    ) {
-        updateStateWithVerificationCodeItems(
-            authenticatorData = authenticatorData.data,
-            clearDialogState = false,
-        )
+    private fun handleDownloadBitwardenClick() {
+        sendEvent(ItemListingEvent.NavigateToBitwardenListing)
     }
 
-    private fun authenticatorDataLoadingReceive() {
+    private fun handleDownloadBitwardenDismiss() {
+        settingsRepository.hasUserDismissedDownloadBitwardenCard = true
         mutableStateFlow.update {
             it.copy(
-                viewState = ItemListingState.ViewState.Loading,
-            )
-        }
-    }
+                viewState = when (it.viewState) {
+                    ItemListingState.ViewState.Loading -> it.viewState
+                    is ItemListingState.ViewState.Content -> it.viewState.copy(
+                        actionCard = ItemListingState.ActionCardState.None,
+                    )
 
-    private fun authenticatorNoNetworkReceive(
-        state: DataState.NoNetwork<List<VerificationCodeItem>>,
-    ) {
-        if (state.data != null) {
-            updateStateWithVerificationCodeItems(
-                authenticatorData = state.data,
-                clearDialogState = true,
+                    is ItemListingState.ViewState.NoItems -> it.viewState.copy(
+                        actionCard = ItemListingState.ActionCardState.None,
```

Comment:
Can we get a test for this state?

Reply:
Yep, thanks for the reminder! 5534b32446cf6fedb44e38d767186cc4ca94712e